File: owner_platform/src/lib/bookingLabels.ts
import type { TFunction } from 'i18next';

/** Aligns with mobile `booking_labels.dart` — maps API codes to owner UI strings. */
export function localizedBookingStatus(t: TFunction, raw: string): string {
  const key = raw.toLowerCase().trim();
  const map: Record<string, string> = {
    created: 'booking.status.created',
    confirmed: 'booking.status.confirmed',
    completed: 'booking.status.completed',
    cancelled: 'booking.status.cancelled',
    pending: 'booking.status.pending',
    hold: 'booking.status.hold',
    active: 'booking.status.active',
  };
  const trKey = map[key];
  return trKey ? t(trKey) : humanizeCode(raw);
}

export function localizedPaymentStatus(t: TFunction, raw: string): string {
  const s = raw.toLowerCase().replaceAll(' ', '_');
  const map: Record<string, string> = {
    paid: 'booking.payment.paid',
    succeeded: 'booking.payment.paid',
    pending: 'booking.payment.pending',
    hold: 'booking.payment.hold',
    failed: 'booking.payment.failed',
    refunded: 'booking.payment.refunded',
    completed: 'booking.payment.completed',
  };
  const trKey = map[s];
  return trKey ? t(trKey) : humanizeCode(raw);
}

export function localizedCancelReason(t: TFunction, raw: string): string {
  const normalized = raw.trim().toLowerCase().replaceAll(' ', '_');
  const map: Record<string, string> = {
    payment_timeout: 'booking.cancelReason.paymentTimeout',
    user_cancelled: 'booking.cancelReason.userCancelled',
    cancelled_by_user: 'booking.cancelReason.userCancelled',
    expired: 'booking.cancelReason.expired',
    hold_expired: 'booking.cancelReason.holdExpired',
    admin_cancelled: 'booking.cancelReason.adminCancelled',
    cancelled_by_admin: 'booking.cancelReason.adminCancelled',
    refund: 'booking.cancelReason.refund',
    refunded: 'booking.cancelReason.refund',
  };
  const trKey = map[normalized];
  if (trKey) return t(trKey);
  if (normalized === 'cancelled by user') return t('booking.cancelReason.userCancelled');
  return humanizeCode(raw);
}

function humanizeCode(raw: string): string {
  return raw
    .replaceAll('_', ' ')
    .split(' ')
    .filter(Boolean)
    .map((w) => (w.length ? w[0].toUpperCase() + w.slice(1).toLowerCase() : ''))
    .join(' ');
}

File: frontend/lib/features/reservations/presentation/booking_labels.dart
import '../../../../l10n/app_localizations.dart';

String localizedBookingStatus(AppLocalizations l10n, String raw) {
  switch (raw.toLowerCase().trim()) {
    case 'created':
      return l10n.bookingStatusCreated;
    case 'confirmed':
      return l10n.confirmed;
    case 'completed':
      return l10n.completed;
    case 'cancelled':
      return l10n.cancelled;
    case 'pending':
      return l10n.bookingStatusPending;
    case 'hold':
      return l10n.bookingStatusHold;
    case 'active':
      return l10n.bookingStatusActive;
    default:
      return _humanizeCode(raw);
  }
}

String localizedCancellationReason(AppLocalizations l10n, String raw) {
  final normalized = raw.trim().toLowerCase().replaceAll(' ', '_');
  switch (normalized) {
    case 'payment_timeout':
      return l10n.cancelReasonPaymentTimeout;
    case 'user_cancelled':
    case 'cancelled_by_user':
      return l10n.cancelReasonUserCancelled;
    case 'expired':
    case 'hold_expired':
      return l10n.cancelReasonHoldExpired;
    case 'admin_cancelled':
    case 'cancelled_by_admin':
      return l10n.cancelReasonAdminCancelled;
    case 'refund':
    case 'refunded':
      return l10n.cancelReasonRefunded;
    case 'minimum_not_met':
      return l10n.cancelReasonMinimumNotMet;
    case 'owner_cancelled':
      return l10n.cancelReasonOwnerCancelled;
    case 'session_expired':
      return l10n.cancelReasonSessionExpired;
    case 'hold_creation_failed':
      return l10n.cancelReasonHoldCreationFailed;
    default:
      if (raw.trim().toLowerCase() == 'cancelled by user') {
        return l10n.cancelReasonUserCancelled;
      }
      return _humanizeCode(raw);
  }
}

String _humanizeCode(String raw) {
  return raw
      .replaceAll('_', ' ')
      .split(' ')
      .map((w) => w.isEmpty
          ? ''
          : '${w[0].toUpperCase()}${w.length > 1 ? w.substring(1).toLowerCase() : ''}',)
      .join(' ');
}

String localizedPaymentStatus(AppLocalizations l10n, String raw) {
  final s = raw.toLowerCase().replaceAll(' ', '_');
  switch (s) {
    case 'paid':
    case 'succeeded':
      return l10n.paymentStatusPaid;
    case 'pending':
      return l10n.paymentStatusPending;
    case 'hold':
      return l10n.paymentStatusHold;
    case 'failed':
      return l10n.paymentStatusFailed;
    case 'refunded':
      return l10n.paymentStatusRefunded;
    case 'completed':
      return l10n.completed;
    default:
      return raw.replaceAll('_', ' ').replaceFirstMapped(
            RegExp(r'^[a-z]'),
            (m) => m.group(0)!.toUpperCase(),
          );
  }
}
